tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+		backupContexts                   []*BackupRestoreContext // Each Context is for one backup
+		restoreContexts                  []*BackupRestoreContext // Each Context is for one restore
+		restoreLaterContexts             []*BackupRestoreContext // Each Context is for one restore-later
+		preRuleNameList                  []string
+		postRuleNameList                 []string
+		appContexts                      []*scheduler.Context // All app contexts (namespaces) in this loop of the scaling
+		sourceNamespaces                 []string
+		destinationNamespaces            []string
+		namespaceMapping                 map[string]string
+		clusterUid                       string
+		clusterStatus                    api.ClusterInfo_StatusInfo_Status
+		cloudCredName                    string
+		cloudCredUID                     string
+		backupLocationUID                string
+		bkpLocationName                  string
+		srcMaj, srcMin, destMaj, destMin int64
+	)
+
+	backupLocationMap := make(map[string]string)
+	labelSelectors := make(map[string]string)
+	sourceNamespaces = make([]string, 0)
+	destinationNamespaces = make([]string, 0)
+	namespaceMapping = make(map[string]string)
+	providers := getProviders()
+
+	JustBeforeEach(func() {
+
+		StartTorpedoTest("BkpRstrDiffK8sVerSimultaneousDiffNS", "Backup on NS(=yy), K8-version(=x) ;; Restore on NS=yy, K8s-version=x+?(=z) [success]; Simultaneously, Restore on K8s-version=z, NS=abc [partial success]", nil, 0)
+
+		log.InfoD("Verifying if the pre/post rules for the required apps are present in the AppParameters or not ")
+		for i := 0; i < len(appList); i++ {
+			if Contains(postRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["post"]; ok {
+					dash.VerifyFatal(ok, true, "Post Rule details mentioned for the apps")
+				}
+			}
+			if Contains(preRuleApp, appList[i]) {
+				if _, ok := portworx.AppParameters[appList[i]]["pre"]; ok {
+					dash.VerifyFatal(ok, true, "Pre Rule details mentioned for the apps")
+				}
+			}
+		}
+
+	})
+
+	It("Backup on NS(=yy), K8-version(=x) ;; Restore on NS=yy, K8s-version=x+?(=z) [success]; Simultaneously, Restore on K8s-version=z, NS=abc [partial success]", func() {
+
+		Step("Verify if 'test app spec'", func() {
+
+			log.InfoD("Allowed apps are %v", appsWithCRDsAndWebhooks)
+			for i := 0; i < len(appList); i++ {
+				contains := Contains(appsWithCRDsAndWebhooks, appList[i])
+				dash.VerifyFatal(contains, true,
+					fmt.Sprintf("check if app [%s] can be used as a dummy spec for *this* test", appList[i]))
+			}
+		})
+
+		Step("Verify K8s version of Src and Dest Cluster", func() {
+			log.InfoD("Verify K8s version of Src and Dest Cluster")
+
+			Step("Register cluster for backup", func() {
+				log.InfoD("Register cluster for backup")
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
```

Comment:
That is not true. If you look at the functions, you'll find that **both of them** will trigger an error with ginkgo/gomega which will ensure that `aftersuite` runs.

In fact, if one observes the way those two functions are used, `dash.Verify*` is used when the intention is to indicate that a particular test result that is part test succeeded or failed, whereas `log.FailOnError` is used to show that one of the intermediate steps failed. The intermediate steps aren't the things that are actually being tested, the `dash.Verify*` are the actual test verification.

Reply:
I documented this stuff, I'll share it on a confluence link later. 

yes, that's what `dash.Verify*` does. And it is only used to indicate the passing/failed of test (as per how others have used it). `log.FailOnError` is when an intermediate step, as part of the test, fails. Just because an intermediate step fails, doesn't mean the test in actually failing for that particular product. It just means the testing code (torpedo) failed; not the thing being tested.

Reply:
✅